qt_configure.bzl

```diff
@@ -13,37 +13,70 @@ def qt_autoconf_impl(repository_ctx):
        repository_ctx: repository context
     """
     os_name = repository_ctx.os.name.lower()
-    is_linux_machine = False
-    if os_name.find("windows") != -1:
+    if os_name.find("windows") == -1 and os_name.find("linux") == -1:
+        fail("Unsupported OS: %s" % os_name)
+    is_linux_machine = os_name.find("windows") == -1
+
+    # Location of the different Qt artifacts on Linux system installed standalone as well as Windows
+    # Qt includes
+    # <location>/include
+    #
+    # Qt libraries
+    # <location>/lib
+    #
+    # uic, rcc, moc
+    # <location>/bin
+
+    # Location of the different Qt artifacts on Linux system installed with apt
+    # Qt includes
+    # /usr/include/x86_64-linux-gnu/qt5
+    #
+    # Qt libraries
+    # /lib/x86_64-linux-gnu/
+    # /usr/lib/x86_64-linux-gnu/
+    #
+    # uic, rcc, moc
+    # /usr/bin/
+
+    # If QT_DIR is defined and it exists, we consider that it is a standalone installation
+    # Only in the case that is linux and QT_DIR is not defined it is considerd a none standalone
+    # installation
+    qt_dir_path = _get_env_var(repository_ctx, "QT_DIR", None)
```

Comment:
Is there a reason to not use https://docs.bazel.build/versions/main/skylark/lib/dict.html#get ? It is available at least from 1.0 onward.

Because this uses `QT_DIR`, I think `repository_rule()` down below should set `environ = ["QT_DIR"]` https://docs.bazel.build/versions/main/skylark/lib/globals.html#repository_rule

Reply:
As it is right now you are right that get could be used. What happened is that the code evolved. Initially getting the variable was done in a case insensitive way this is why get could have not been used. This should be redone and make clear what behavior is wanted, here the full discussion: https://github.com/justbuchanan/bazel_rules_qt/pull/17#issuecomment-784937229